sklearn/metrics/cluster/tests/test_common.py

```diff
@@ -0,0 +1,171 @@
+        metric = SUPERVISED_METRICS[name]
+        assert_not_equal(metric(y1, y2), metric(y2, y1))
+
+
+def test_normalized_output():
+    upper_bound_1 = [0, 0, 0, 1, 1, 1]
+    upper_bound_2 = [0, 0, 0, 1, 1, 1]
+    for name in METRICS_NORMALIZED_OUTPUT:
+        metric = SUPERVISED_METRICS[name]
+        assert_between(metric([0, 0, 0, 1, 1], [0, 0, 0, 1, 2]), 0.0, 1.0)
+        assert_between(metric([0, 0, 1, 1, 2], [0, 0, 1, 1, 1]), 0.0, 1.0)
+        assert_equal(metric(upper_bound_1, upper_bound_2), 1.0)
+
+    lower_bound_1 = [0, 0, 0, 0, 0, 0]
+    lower_bound_2 = [0, 1, 2, 3, 4, 5]
+    for name in SUPERVISED_METRICS:
+        metric = SUPERVISED_METRICS[name]
+        assert_equal((metric(lower_bound_1, lower_bound_2) and
+                      metric(lower_bound_2, lower_bound_1)), 0.0)
+
+
+# All clustering metrocs do not change score due to permutations of labels
+# that is when 0 and 1 exchchanged.
+def test_permute_labels():
+    for name in SUPERVISED_METRICS:
+        metric = SUPERVISED_METRICS[name]
+        y_label = np.array([0, 0, 0, 1, 1, 0, 1])
+        y_pred = np.array([1, 0, 1, 0, 1, 1, 0])
+        score_1 = metric(y_pred, y_label)
+        assert_equal(score_1, metric(1 - y_pred, y_label))
+        assert_equal(score_1, metric(1 - y_pred, 1 - y_label))
+        assert_almost_equal(score_1, metric(y_pred, 1 - y_label))
+
+    for name in UNSUPERVISED_METRICS:
+        metric = UNSUPERVISED_METRICS[name]
+        X = np.random.randint(10, size=(100, 10))
+        y = np.random.randint(2, size=100)
+        score_1 = metric(X, y)
+        assert_almost_equal(score_1, metric(X, 1 - y))
+
+
+# For ALL clustering metrics Input parameters can be both
+# in the form of arrays lists , positive , negetive or string
+def test_format_invariance():
+    for name in SUPERVISED_METRICS:
+        metric = SUPERVISED_METRICS[name]
+        list_a = [0, 0, 0, 1, 1, 1]
+        list_b = [0, 1, 2, 3, 4, 5]
+        arr_a = np.array(list_a)
+        arr_b = np.array(list_b)
+        str_a = [str(x) for x in arr_a]
+        str_b = [str(x) for x in arr_b]
+        score_list = metric(list_a, list_b)
+        score_neg = metric(arr_a - 1, arr_b)
+        score_pos = metric(arr_a + 1, arr_b)
+        score_array = metric(arr_a, arr_b)
+        score_str = metric(str_a, str_b)
+        assert_equal(score_list, score_array)
+        assert_equal(score_array, score_neg)
+        assert_equal(score_array, score_pos)
+        assert_equal(score_list, score_str)
+
+    for name in UNSUPERVISED_METRICS:
+        metric = UNSUPERVISED_METRICS[name]
+        X = np.random.randint(10, size=(100, 10))
+        X_array = np.array(X)
+        label = np.random.randint(2, size=100)
+        label_array = np.array(label)
+        X_float = X_array.astype(float)
+        score_1 = metric(X, label)
+        score_2 = metric(X_array, label_array)
+        score_3 = metric(X_float, label_array)
+        score_4 = metric(X_array + 1, label_array)

```

Comment:
this is not appropriate for `X`.

however, we should be checking the same sorts of invariances for `label` as for `*_a` and `*_b` above. I would go as far as to say you should consider using something like:
```
def generate_formats(y):
    y = np.array(y)
    yield y, 'array of ints'
    yield y.tolist(), 'list of ints'
    yield [str(x) for x in y.tolist()], 'list of strs'
    yield y - 1, 'including negative ints'
    yield y + 1, 'strictly positive ints'
```

in both supervised and unsupervised tests (but not applied to `X`) to avoid repeated code.